incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -503,20 +538,50 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	crds := []*unstructured.Unstructured{}
+	crd := &unstructured.Unstructured{}
+	crd.SetGroupVersionKind(crdGVK)
+	for _, crdName := range r.crdNames {
+		crdKey := types.NamespacedName{Name: crdName}
+		if err := r.client.Get(r.ctx, crdKey, crd); err != nil {
+			if k8sErrors.IsNotFound(err) {
+				// Object not found, return.  Created objects are automatically garbage collected.
+				// For additional cleanup logic use finalizers.
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{Requeue: true}, err
+		}
+		crds = append(crds, crd)
+	}
+
 	if secret.GetDeletionTimestamp().IsZero() {
```

Comment:
There's no "else" here, so maybe turn this into an early exit, and then collect all the new CRD stuff (loading the existing CRDs, modifying them and writing them back) into a new function that gets called after line 571?

Reply:
Further split the logic in `Reconcile()` into 3 parts:
1. Get secret and generate certs
2. Ensure certs on validatingwebhookconfiguration
3. Ensure certs on CRDs

Reply:
As I mentioned in my other comments, we should have as little refactoring as possible since that makes it very difficult to upstream this code. My suggestion (collect all the new stuff into one function and call it on line 571) would make it easier to merge these changes in the future.

Reply:
Got you yes, that makes sense. Do you think I can make minimum change in certs.go and still have it with this PR?

Reply:
Hmm, maybe try it out and see?